### DISCUSSION TITLE
OpenAI + Local embeddings

### DISCUSSION THREAD

**Discussion by juffis** [logancyang/obsidian-copilot#173] [Q&A]
Hello!

Is it possible to use OpenAI with local embeddings? I'm not totally sure how these work so this might be a stupid question.

**Comment by logancyang** [comment]
@juffis Yes you should be able to, just choose OpenAI models in the models dropdown, and then pick LocalAI in the embedding provider dropdown in settings, provided that you can LocalAI set up and running correctly with an embedding model. 

<img width="544" alt="Screenshot 2023-11-12 at 3 34 39 PM" src="https://github.com/logancyang/obsidian-copilot/assets/4860545/7d667aa0-1b49-4200-8510-2f1d2db23133">
<img width="797" alt="Screenshot 2023-11-12 at 3 34 55 PM" src="https://github.com/logancyang/obsidian-copilot/assets/4860545/3928467e-9885-412b-b7c0-8f1e57723aa1">

Let me know if you make it work, if yes, does the local embedding model work well. In this [video demo](https://youtu.be/3yPVDI8wZcI?si=JoWJ-oS3cniCk3X-) I created, the local embedding model was just for demo purposes and performed worse than OpenAI embeddings.
